UnitTests/OIDAuthStateTests.m

```diff
@@ -435,6 +435,21 @@ - (void)testIsTokenFreshHandlesTokenWithoutExpirationTime {
   XCTAssertEqual([authState isTokenFresh], YES, @"");
 }
 
+- (void)testThatRefreshTokenExceptionWillBeRaisedForTokenRequestWithAdditionalParameters {
+  OIDAuthState *authState = [[OIDAuthState alloc] initWithAuthorizationResponse:nil tokenResponse:nil registrationResponse:nil];
+  XCTAssertThrowsSpecificNamed([authState tokenRefreshRequestWithAdditionalParameters:nil], NSException, @"Attempted to create a token refresh request from a token response with no refresh token.");
+}
+
+- (void)testThatRefreshTokenExceptionWillBeRaisedForTokenRequestWithAdditionalHeaders {
+  OIDAuthState *authState = [[OIDAuthState alloc] initWithAuthorizationResponse:nil tokenResponse:nil registrationResponse:nil];
+  XCTAssertThrowsSpecificNamed([authState tokenRefreshRequestWithAdditionalHeaders:nil], NSException, @"Attempted to create a token refresh request from a token response with no refresh token.");
```

Comment:
I think it'd be preferable to use [`kRefreshTokenRequestException`](https://github.com/openid/AppAuth-iOS/blob/master/Source/AppAuthCore/OIDAuthState.m#L61) for the argument checking the exception name. That way, we could change the value of the name (if we wanted), and the tests wouldn't automatically fail. You probably need to declare `kRefreshTokenRequestException` in `OIDAuthState.h`.